core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -216,21 +229,80 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
     } yield clone
 
   def cloneAzureWorkspace(sourceWorkspace: Workspace,
-                          project: RawlsBillingProject,
                           profile: ProfileModel,
-                          request: WorkspaceRequest
-  ): Future[Workspace] = Future.failed(
-    RawlsExceptionWithErrorReport(
-      ErrorReport(StatusCodes.NotImplemented, "WOR-625")
-    )
-  )
+                          request: WorkspaceRequest,
+                          parentContext: RawlsRequestContext
+  ): Future[Workspace] = {
+
+    def createNewWorkspaceRecord(): Future[Workspace] =
+      dataSource.inTransaction { access =>
+        for {
+          _ <- failIfWorkspaceExists(request.toWorkspaceName)
+          workspaceId = UUID.randomUUID()
+          newWorkspace <- createMultiCloudWorkspaceInDatabase(
+            workspaceId.toString,
+            request.toWorkspaceName,
+            request.attributes,
+            access,
+            parentContext
+          )
+        } yield newWorkspace
+      }
+
+    for {
+      // The call to WSM is asynchronous. Before we fire it off, allocate a new workspace record
+      // to avoid naming conflicts - we'll erase it should the clone request to WSM fail.
+      newWorkspace <- createNewWorkspaceRecord()
+
+      cloneResult <- traceWithParent("workspaceManagerDAO.cloneWorkspace", parentContext) { context =>
```

Comment:
You need to check the `cloneResult` to see if it included an ErrorReport. If so, you need to delete the Rawls workspace record (if you've already created it… I still think you should wait to create until you know that the clone was kicked off successfully).

I had a previous clone attempt that failed due, and now I'm seeing this, indicating that the Rawls record wasn't deleted.

![image](https://user-images.githubusercontent.com/484484/208929947-65f099d5-e2ef-45ba-8113-1dfe0f943c62.png)

Reply:
This was the error I hit when trying to clone the first time. I thought it was because my branch was out-of-date, but I pulled and am still seeing it:

![image](https://user-images.githubusercontent.com/484484/208930398-603cf8bf-deec-4a53-9f16-d5620f44f5ad.png)


{"causes":[],"exceptionClass":"java.lang.IllegalArgumentException","message":"Invalid UUID string: 1MJcwag2S4WZiGfDqKnjuA","source":"rawls","stackTrace":[{"className":"java.util.UUID","fileName":"UUID.java","lineNumber":280,"methodName":"fromString1"},{"className":"java.util.UUID","fileName":"UUID.java","lineNumber":258,"methodName":"fromString"},{"className":"org.broadinstitute.dsde.rawls.workspace.MultiCloudWorkspaceService","fileName":"MultiCloudWorkspaceService.scala","lineNumber":290,"methodName":"$anonfun$cloneAzureWorkspace$7"},{"className":"scala.concurrent.impl.Promise$Transformation","fileName":"Promise.scala","lineNumber":470,"methodName":"run"},{"className":"java.util.concurrent.ForkJoinTask$RunnableExecuteAction","fileName":"ForkJoinTask.java","lineNumber":1395,"methodName":"exec"},{"className":"java.util.concurrent.ForkJoinTask","fileName":"ForkJoinTask.java","lineNumber":373,"methodName":"doExec"},{"className":"java.util.concurrent.ForkJoinPool$WorkQueue","fileName":"ForkJoinPool.java","lineNumber":1182,"methodName":"topLevelExec"},{"className":"java.util.concurrent.ForkJoinPool","fileName":"ForkJoinPool.java","lineNumber":1655,"methodName":"scan"},{"className":"java.util.concurrent.ForkJoinPool","fileName":"ForkJoinPool.java","lineNumber":1622,"methodName":"runWorker"},{"className":"java.util.concurrent.ForkJoinWorkerThread","fileName":"ForkJoinWorkerThread.java","lineNumber":165,"methodName":"run"}]}

Reply:
I'm going to let the runner take care of that. It'll already have to to handle this error so it makes sense to do it in one place rather than duplicating this logic.

Reply:
I do think you want to at least verify if you get a JobResult with a JobID in this method. I'm pretty sure we have seen cases in other WSM methods where we get an ErrorReport and no JobID because the operation immediately failed.

Reply:
If we get an error report why is the request responding with a non error code in the http response? That's really weird - akin to mixing error codes and exceptions when people objected to exceptions "for performance reasons"